component/binary_info.go

```diff
@@ -14,20 +14,24 @@
 
 package component
 
-// ApplicationStartInfo is the information that is logged at the application start and
+// BinaryInfo is the information that is logged at the application start and
 // passed into each component. This information can be overridden in custom builds.
-type ApplicationStartInfo struct {
+type BinaryInfo struct {
```

Comment:
`BuildInfo` sounds good to me.  I don't think that `debug.BuldInfo.Path` works for our present use case, but I can definitely see where having that identifier that uniquely identifies the package and can be programmatically inserted into the `BuildInfo` is useful.  Someone may copypasta the `BuildInfo` definition from this repo and not update the description, but they wouldn't be able to avoid having the toolchain report the package path it was built from.  I'll work with @dhruv-vora to get a follow-up issue/PR created for that.